vowpalwabbit/search_graph.cc

```diff
@@ -83,10 +85,14 @@ struct task_data
   vector<size_t>   pred;  // predictions
   example*cur_node;       // pointer to the current node for add_edge_features_fn
   float* neighbor_predictions;  // prediction on this neighbor for add_edge_features_fn
-  weight* weight_vector;
+  array_parameters weight_vector;
   uint32_t* confusion_matrix;
   float* true_counts;
   float true_counts_total;
+
+  task_data() :weight_vector(nullptr)
```

Comment:
in general I'd not assume constructors being called.
most of the structs are allocated using calloc()